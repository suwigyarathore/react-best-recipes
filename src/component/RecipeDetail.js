import classNames from 'classnames'

const RecipeDetail = (props) => {
  return props.recipe ? (
    <div style={props.style} className={classNames(`p2 bg-white ${ props.className }`)}>
      <h2 className="h2">{props.recipe.name}</h2>
      <img className="fit" src={props.recipe.image}></img>
      <div>
        <span>{props.recipe.category}</span>
        <span>{props.recipe.calories}</span>
      </div>
      <h3>Ingrediends</h3>
      <ul>
        {props.recipe.ingredients.map((ingredient, i) => (
          <li key={i}>
            {ingredient}
          </li>
        ))}
      </ul>
      <h3>Steps</h3>
      <ol>
        {props.recipe.steps.map((step, i) => (
          <li key={i}>
            {step}
          </li>
        ))}
      </ol>
    </div>
  ) :
    (<p style={props.style} className={classNames(`h3 p2 bg-white italic center ${ props.className }`)}>
      Please select recipe to see detail
  </p>);
}

export default RecipeDetail;
