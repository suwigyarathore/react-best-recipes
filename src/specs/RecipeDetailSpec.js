import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RecipeDetail from '../component/RecipeDetail';

describe('<RecipeDetail/>', () => {
  let testRecipe;
  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test Recipe',
      category: 'Test Category',
      ingredients: ['i1', 'i2'],
      steps: ['s1', 's2'],
    };
  });

  test('Should render zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly a recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly assign a classname', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className="classname-test" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipe without ingredient', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipe without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
