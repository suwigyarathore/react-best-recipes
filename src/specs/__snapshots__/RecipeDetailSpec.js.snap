// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail/> Should correctly assign a classname 1`] = `
<div
  className="p2 bg-white classname-test"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingrediends
  </h3>
  <ul>
    <li>
      i1
    </li>
    <li>
      i2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      s1
    </li>
    <li>
      s2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail/> Should render correctly a recipe 1`] = `
<div
  className="p2 bg-white undefined"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingrediends
  </h3>
  <ul>
    <li>
      i1
    </li>
    <li>
      i2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      s1
    </li>
    <li>
      s2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail/> Should render recipe without ingredient 1`] = `
<div
  className="p2 bg-white undefined"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingrediends
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      s1
    </li>
    <li>
      s2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail/> Should render recipe without steps 1`] = `
<div
  className="p2 bg-white undefined"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    className="fit"
  />
  <div>
    <span>
      Test Category
    </span>
    <span />
  </div>
  <h3>
    Ingrediends
  </h3>
  <ul>
    <li>
      i1
    </li>
    <li>
      i2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail/> Should render zero state 1`] = `
<p
  className="h3 p2 bg-white italic center undefined"
>
  Please select recipe to see detail
</p>
`;
